packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -527,9 +527,24 @@ module.exports = async (api, args = {}, { pluginSource, activity } = {}) =>
         return null
       }
 
+      let gatsbyNode = pluginNodeCache.get(plugin.name)
+      if (!gatsbyNode) {
+        gatsbyNode = require(`${plugin.resolve}/gatsby-node`)
+        pluginNodeCache.set(plugin.name, gatsbyNode)
+      }
+
       const pluginName =
         plugin.name === `default-site-plugin` ? `gatsby-node.js` : plugin.name
 
+      if (
+        api === `onCreateNode` &&
+        gatsbyNode?.onCreateNodeSyncTest && // Don't bail if this api is not exported
+        !gatsbyNode.onCreateNodeSyncTest(args.node)
+      ) {
+        // Do not try to schedule an async event for this node for this plugin
+        return null
+      }
```

Comment:
```suggestion
      // TODO rethink createNode API to handle this better
      if (
        api === `onCreateNode` &&
        gatsbyNode?.onCreateNodeSyncTest && // Don't bail if this api is not exported
        !gatsbyNode.onCreateNodeSyncTest(args.node)
      ) {
        // Do not try to schedule an async event for this node for this plugin
        return null
      }
```

Reply:
Just double checking; the only suggestion here is the comment on top, right?

Reply:
yeah the todo